### PR TITLE
Zachmu/xx

### PR DESCRIPTION
**Pull request by zachmu**
Use xxhash everywhere, and standardize the construction of hash keys.

### CODE REVIEW

File: sql/plan/group_by.go [hunk 1]
```diff
@@ -315,25 +315,24 @@ func (i *groupByGroupingIter) Close() error {
...
 			return 0, err
 		}
-		vals = append(vals, fmt.Sprintf("%#v", v))
+		_, err = hash.Write(([]byte)(fmt.Sprintf("%#v,", v)))
+		if err != nil {
+			return 0, err
+		}
 	}
 
-	// TODO: use a faster hash func
```
Comment (andrew-wm-arthur) [inline sql/plan/group_by.go:335]: you did it!

File: sql/cache.go [hunk 2]
```diff
@@ -2,19 +2,25 @@ package sql
...
 	errors "gopkg.in/src-d/go-errors.v1"
 )
 
-var table = crc64.MakeTable(crc64.ISO)
-
 // CacheKey returns a hash of the given value to be used as key in
 // a cache.
-func CacheKey(v interface{}) uint64 {
-	return crc64.Checksum([]byte(fmt.Sprintf("%#v", v)), table)
+func CacheKey(v Row) (uint64, error) {
```
Comment (andrew-wm-arthur) [inline sql/cache.go:15]: what about just `Hash()` or `HashOf`. there are lots of ways to key into a cache

**Review by andrew-wm-arthur — APPROVED** [review]
lgtm

File: sql/cache.go (hunk 2 above)
Comment (zachmu) [reply sql/cache.go]: Done

**Pull request merged**